### ISSUE TITLE
breaking change in default behaviour I'm relaying on and seems relevant property it's not configurable

### ISSUE THREAD

**Issue by n0npax** [OpenSLO/oslo#194]
### Summary

Please consider [PR-187](https://github.com/OpenSLO/oslo/pull/187) as relevant source.

Within nobl9 indicator kind keyword `"Direct"` was replaced by `"Agent"`. This changes default `oslo` behaviour I was relying on.

old code was assuming given hardcode:

```
		return nobl9v1alpha.Indicator{
			MetricSource: nobl9v1alpha.MetricSourceSpec{
				Kind:    "Direct", # <---------
				Name:    rmt,
				Project: project,
			},
```
but new one is
```
			return nobl9v1alpha.Indicator{
				MetricSource: nobl9v1alpha.MetricSourceSpec{
					Project: metricSourceProject,
					Name:    name,
					Kind:    "Agent", # <----------
				},
```

Please also consider given [codeblck](https://github.com/OpenSLO/oslo/pull/187/files#diff-8f9b223b72569b1b5a758d2828bf9e6f0b6e357afb8154fc2ddd59df3df71ed6R235-R240):

```
	// Since we don't have a way of specifying MetricSource.Kind in OpenSLO, use Nobl9's default
	// of Agent, and warn the user
	_ = printWarning(
		"We don't have a way of specifying the MetricSource Kind (Agent or Direct) in OpenSLO " +
			"so we will use Nobl9's default of Agent",
	)
```

With current state I believe I can't adjust my config to change `kind` type on demand and new default value doesn't work for me.

### Steps to reproduce

Convert `openSLO` to `nobl9` SLO within splunk based metrics.
```sh
 oslo convert -f ${INPUT_SLO}.yaml -f ${INPUT_SERVICE}.yaml -o nobl9 -p llama
```

### What is the current *bug* behavior?

Tool generates invalid nobl9 SLO and warns user.

### What is the expected *correct* behavior?

- [ ] Ability to specify `"Direct"` or `"Agent"` on demand (preferred)
- [ ] Change of default value to `"Direct"` (less preferred)

### Relevant logs and/or screenshots

Sample relevant piece of output I can generate looks like:

```
---
apiVersion: n9/v1alpha
kind: SLO
metadata:
  name: foo-availability
  project: llama
  ...
spec:
  indicator:
    metricSource:
      name: ChangeMe # <—---- Problem with generated output is here
      kind: ""
     ....
```

using old version of `oslo`(docker image with tag `sha-0ef3a32`) I'm able to get indicator value with expected data.

```
      indicator:
          metricSource:
              project: redacted
              name: redacted
              kind: Direct
```

Redacted `slo` which can be used to test behaviour looks like: 

```
apiVersion: openslo/v1
kind: SLO
metadata:
  name: redacted
  displayName: redacted [availability]
  # labels:
    # redacted
  #annotations:
    # redacted
spec:
  description: 'redacted http://example.com/FIXME'
  service: redacted
  indicator:
    metadata:
      name: redacted
      labels: {}
      annotations: {}
    spec:
      ratioMetric:
        counter: true
        good:
          metricSource:
            metricSourceRef: redacted
            type: Splunk
            spec:
              query: redacted
        total:
          metricSource:
            metricSourceRef: redacted
            type: Splunk
            spec:
              query: redacted
  budgetingMethod: Occurrences
  timeWindow:
    - duration: 28d
      isRolling: true
  objectives:
    - displayName: good
      value: 1
      target: 0.987654321
  alertPolicies: []
  ```

**Comment by fpiwowarczyk** [comment]
Thanks for your submission! It will assign myself to this issue and try to do something with that one.

**Comment by fpiwowarczyk** [comment]
Changing to direct again doesn't make sense for me, it will be problematic for someone else in future. 

I'd like to go for to specifying  kind of indicator. First that comes to my mind is to add flag like `--kind Agent` but need to investigate if it makes sens and won't be problematic in future. 

Any preferred way @n0npax ? :D

**Comment by n0npax** [comment]
flag would work, but it would make usage a bit problematic though. More flags user has to pass, user experience is getting worse. As a user I prefer to keep setup just within yaml file and I'd prefer avoiding leaks from SLO configs to the deployment script.

Would it be possible to extend openSLO spec to include the `kind` or react to a specific/magic annotation?
At the end of the day I'll be generating vast majority of openSLOs via exposed `go` structs, so I'd prefer the config to be exposed within the `oslo` structs.

**Comment by fpiwowarczyk** [comment]
I think including `kind` in `OpenSLO` isn't possible because it is Nobl9 internal concept, OpenSLO should stay vendor agnostic and flexible but in same time we don't want to over complicate spec :D. I'll try to read annotation in OpenSLO spec when converting to nobl9 spec. My current idea is to have something like that `#annotation:nobl9-kind:direct/agent` it will allow possible other vendors add their spec annotations converters their internal fields.

**Comment by fpiwowarczyk** [comment]
@n0npax I prepared PR with possible solution of your issue, if you have some time you could tell me if that suits you :business_suit_levitating:

**Comment by n0npax** [comment]
Thanks for help 🚀 
I left single comment on PR. Feel free to catch me on a slack if you wish to discuss it further :)

**Comment by fpiwowarczyk** [comment]
Closing it as resolved by #199
